zest/releaser/utils.py

```diff
@@ -494,6 +518,47 @@ def system(command, input=''):
     return result
 
 
+def retry_command(command):
+    """Run the command and possibly retry it.
+
+    Print interesting lines.
+
+    When we discover errors, we give three options:
+    - Abort
+    - Retry
+    - Continue
+
+    There is an error is there is a red color in the output.
+
+    It might be a warning, but we cannot detect the distinction.
+    """
+    result = system(command)
+    if Fore.RED not in result:
+        show_interesting_lines(result)
+        return result
+    # There are warnings or errors. Print the complete result.
+    print(result)
+    question = ("There were errors or warnings. Are you sure "
+                "you want to continue? [Yes/No/Retry]")
+    if AUTO_RESPONSE:
+        msg = ("The question '%s' requires a manual answer, but "
+               "we're running in --no-input mode.")
+        msg = msg % question
+        raise RuntimeError(msg)
+    while True:
+        input = get_input(question)
+        if input:
+            input = input.lower()
+            if input == 'y':
+                # Accept the error, continue with the program.
+                return result
+            if input == 'n':
+                sys.exit(1)
```

Comment:
Well, we can stick to yes/no/retry. But then we need to print a little bit of extra info on the result of the choice. Something like this, perhaps less verbose:

Continue ('y'es): we have placed a fresh tag checkout in /tmp/dfdfsdfsdf, you can re-try uploading from there if needed.

Retry: do this if it looks like a temporary internet or pypi outage.

Quit ('n'o): stop completely. Note that the postrelease step hasn't been run then, you need to do that manually.


Something like that?